pypsa/linopf.py

```diff
@@ -1300,8 +1377,13 @@ def map_solution(c, attr):
             n.solutions.at[(c, attr), "pnl"] = False
             sol = variables.map(variables_sol)
             if predefined:
-                non_ext = n.df(c)[attr]
-                n.df(c)[attr + "_opt"] = sol.reindex(non_ext.index).fillna(non_ext)
+                if not (
+                    attr == "n_opt"
+                ):  # if inserito perchè la seguente riga serve solo per far diventare quello che in pypsa viene ottimizzato come nominal optimal, quindi entravamo in conflitto
```

Comment:
At first glance, I've the feeling that the need for this if is no more necessary after making the change on variables.csv, but needs testing